js/tidebit.js

```diff
@@ -0,0 +1,426 @@
+'use strict';
+
+//  ---------------------------------------------------------------------------
+
+const Exchange = require ('./base/Exchange');
+const { InsufficientFunds, OrderNotFound } = require ('./base/errors');
+// const { ROUND, TRUNCATE, SIGNIFICANT_DIGITS } = require ('./base/functions/number');
+
+//  ---------------------------------------------------------------------------
+
+module.exports = class tidebit extends Exchange {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'id': 'tidebit',
+            'name': 'TideBit',
+            'countries': 'HK',
+            'rateLimit': 1000,
+            'version': 'v2',
+            'has': {
+                'fetchDepositAddress': true,
+                'CORS': true,
+                'fetchTickers': true,
+                'fetchOHLCV': true,
+                'withdraw': true,
+            },
+            'timeframes': {
+                '1m': '1',
+                '5m': '5',
+                '15m': '15',
+                '30m': '30',
+                '1h': '60',
+                '2h': '120',
+                '4h': '240',
+                '12h': '720',
+                '1d': '1440',
+                '3d': '4320',
+                '1w': '10080',
+            },
+            'urls': {
+                'logo': 'https://user-images.githubusercontent.com/629338/37326862-f836f4de-26ce-11e8-93ac-7003b3e8a280.png',
+                'extension': '.json',
+                'api': 'https://www.tidebit.com/api',
+                'www': 'https://www.tidebit.com',
+                'doc': 'https://www.tidebit.com/documents/api_v2',
+            },
+            'api': {
+                'public': {
+                    'get': [
+                        'v2/markets', // V2MarketsJson
+                        'v2/tickers', // V2TickersJson
+                        'v2/tickers/{market}', // V2TickersMarketJson
+                        'v2/trades', // V2TradesJson
+                        'v2/trades/{market}', // V2TradesMarketJson
+                        'v2/order_book', // V2OrderBookJson
+                        'v2/order', // V2OrderJson
+                        'v2/k_with_pending_trades', // V2KWithPendingTradesJson
+                        'v2/k', // V2KJson
+                        'v2/depth', // V2DepthJson
+                    ],
+                    'post': [],
+                },
+                'private': {
+                    'get': [
+                        'v2/deposits', // V2DepositsJson
+                        'v2/deposit_address', // V2DepositAddressJson
+                        'v2/deposit', //  V2DepositJson
```

Comment:
By PEP8 syntax standard Python wants this comment to look like `// V2DepositJson` (one space after the double slash, in your code there's two spaces, Python PEP8 is very critical to that, so it complains about this line and the next one. Everything else looks very clean! Thanks so much!

Reply:
@jjhesk can you plz remove the excessive spaces from those two lines? thx!

Reply:
ok @kroitor

Reply:
@kroitor do i have something else need to work on? those two lines spaces are removed.

Reply:
@jjhesk nope, everything else is ok, I'll do the cleanup and will merge it shortly, no worries. Thanks so much for your effort!

Reply:
im excited to see it on production!